pkg/manifest/mantaray/marshal.go

```diff
@@ -256,13 +256,14 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 			f := &fork{}
 
 			if len(data) < offset+nodeForkPreReferenceSize+refBytesSize {
-				err := fmt.Errorf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize))
-				return fmt.Errorf("%s on byte '%x': %w", err, []byte{b}, ErrInvalidManifest)
+				cause := fmt.Sprintf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize))
+				return fmt.Errorf("%s on byte '%x': %w", cause, []byte{b}, ErrInvalidManifest)
 			}
 
 			err := f.fromBytes(data[offset : offset+nodeForkPreReferenceSize+refBytesSize])
 			if err != nil {
-				return fmt.Errorf("%s on byte '%x': %w", err, []byte{b}, ErrInvalidManifest)
+				cause := err.Error()
```

Comment:
Not sure why the `cause` is factored out. The same is below.

Reply:
refactored